browser/src/react-redux/App/LoginPage.js

```diff
@@ -0,0 +1,96 @@
+import React from 'react'
+import { connect } from 'react-redux'
+import { loginSuccess, loginFailure } from './User/updateUser.action'
+import axios from 'axios'
+
+const initialState = {
+  email: '',
+  password: '',
+  generalError: ''
+}
+
+class LoginPage extends React.Component {
+  state = initialState
+
+  handleChange = event => {
+    this.setState({
+      [event.target.name]: event.target.value
+    })
+  }
+
+  onSubmit = e => {
+    e.preventDefault()
+    this.setState(initialState)
+
+    const user = {
+      email: this.state.email,
+      password: this.state.password
+    }
+
+    axios
+      .post('/api/auth/login', user)
+      .then(res => {
+        this.props.loginSuccess(user)
+      })
+      .catch(error => {
+        this.props.loginFailure(user)
+        const errorMsg = error.response.data.error.message
+        this.setState({ generalError: errorMsg })
+      })
+  }
+  componentDidUpdate() {
+    if (this.props.user.loggedIn === true) {
```

Comment:
Remember, the user in the store should either be null or populated. As mentioned in other comments about `.loggedIn`, that property isn't needed. Let's just check to see if `this.props.user` is populated.